src/Powercord/plugins/pc-translate/index.js

```diff
@@ -61,10 +99,54 @@ module.exports = class Translate extends Plugin {
             [ ...message.querySelectorAll('.pc-markup') ]
               .map(async (markup) => {
                 const markupInstance = getOwnerInstance(markup);
-                const { text, from } = await translate(markupInstance.props.message.contentParsed.filter(c => typeof c === 'string').join(''), opts);
-                _this.translations[markupInstance.props.message.id] = text;
-                fromLang = translate.languages[from.language.iso];
-                markupInstance.forceUpdate();
+                const { embed, message } = markupInstance.props;
+
+                if (embed || (message && message.embeds.length > 0)) {
+                  const embed = markupInstance.props.embed || message.embeds[0];
+                  const { text, from } = await translate(embed.description, opts);
+
+                  _this.translations[embed.id] = text;
+                  fromLang = translate.languages[from.language.iso];
+                }
+
+                let content;
+
+                if (message) {
+                  if (message.contentParsed.length > 1) {
+                    const contentArray = [];
+                    const contentStrings = message.contentParsed.filter(content => typeof content === 'string');
+
+                    contentStrings.map(async (content) => {
+                      const { text, from } = await translate(content, opts);
+
+                      fromLang = translate.languages[from.language.iso];
+                      contentArray.push({ original: content,
+                        translation: content.startsWith(' ') && content.endsWith(' ')
+                          ? ` ${text} `
+                          : content.startsWith(' ') ? ` ${text}` : content.endsWith(' ') ? `${text} ` : text });
+
+                      return contentArray;
+                    });
+
+                    while (contentArray.length !== contentStrings.length) {
+                      await new Promise(resolve => setTimeout(resolve, 100));
+                    }
+
+                    content = contentArray;
+                  } else {
+                    const { text, from } = await translate(
+                      message.contentParsed.filter(content => typeof content === 'string').join(''),
+                      opts
+                    );
+
+                    fromLang = translate.languages[from.language.iso];
+                    content = text;
+                  }
+
+                  _this.translations[message.id] = content;
+                }
+
+                forceUpdateElement('.pc-markup', true);
```

Comment:
With the deprecation of .pc- classes it might be a good idea moving away from thoses here

Reply:
/***resolved***; changes have since been reflected in https://github.com/powercord-org/powercord/pull/226/commits/3c2872246dfed4c380bf1ace4d7ef1c1aeec11db.